parsec/core/fs/workspacefs/workspacefile.py

```diff
@@ -87,7 +87,8 @@ async def ainit(self):
                 _, self._fd = await self._transactions.file_create(self._path, open=True)
             else:
                 try:
-                    _, self._fd = await self._transactions.file_open(self._path)
+                    mode = "rw" if self.writable() else "r"
+                    _, self._fd = await self._transactions.file_open(self._path, mode)
                 except FileNotFoundError:
                     if sum(c in self._mode for c in "aw") == 1:
```

Comment:
I'd rather do
```python
if 'a' in self._mode or 'w' in self._mode
```
(especially given we already checked that we cannot have both 'a' and 'w' at the same time)

beside for the sum, it would be better not to take advantage of the fact `int(True) == 1` given it's more of a legacy detail that a real feature:
```python
        if sum(1 for c in mode if c in "rwax") != 1:
```

Reply:
Yes it seems better, I modify that

Reply:
> it would be better not to take advantage of the fact int(True) == 1 given it's more of a legacy detail that a real feature

Well, `True == 1` and `False == 1` are more than a legacy detail to me, I don't think it's something that could be discarded from the language. But I'm fine with both approaches, just being pedantic here :)